test/e2e/hooks_test.go

```diff
@@ -28,34 +33,90 @@ metadata:
 spec:
   entrypoint: main
   hooks:
-    exit:
-      template: http
     running:
       expression: workflow.status == "Running"
-      template: http
+      template: argosay
+    succeed:
+      expression: workflow.status == "Succeeded"
+      template: argosay
+
   templates:
     - name: main
       steps:
       - - name: step1
-          template: http
+          template: argosay
 
-    - name: http
-      http:
-        url: "http://httpstat.us"
+    - name: argosay
+      container:
+        image: argoproj/argosay:v2
+        command: ["/argosay"]
 `).When().
 		SubmitWorkflow().
 		WaitForWorkflow(fixtures.ToBeSucceeded).
 		Then().
 		ExpectWorkflow(func(t *testing.T, metadata *v1.ObjectMeta, status *v1alpha1.WorkflowStatus) {
 			assert.Equal(t, status.Phase, v1alpha1.WorkflowSucceeded)
 		}).ExpectWorkflowNode(func(status v1alpha1.NodeStatus) bool {
-		return strings.Contains(status.Name, "hook")
-	}, func(t *testing.T, status *v1alpha1.NodeStatus, pod *apiv1.Pod) {
-		assert.Equal(t, v1alpha1.NodeSucceeded, status.Phase)
-	})
+			return strings.Contains(status.Name, ".hooks.running")
+		}, func(t *testing.T, status *v1alpha1.NodeStatus, pod *apiv1.Pod) {
+			assert.Equal(t, v1alpha1.NodeSucceeded, status.Phase)
+		}).ExpectWorkflowNode(func(status v1alpha1.NodeStatus) bool {
+			return strings.Contains(status.Name, ".hooks.succeed")
+		}, func(t *testing.T, status *v1alpha1.NodeStatus, pod *apiv1.Pod) {
+			assert.Equal(t, v1alpha1.NodeSucceeded, status.Phase)
+		})
+}
+
+func (s *HooksSuite) TestWorkflowLevelHooksFailVersion() {
+	s.Given().
+		Workflow(`apiVersion: argoproj.io/v1alpha1
+kind: Workflow
+metadata:
+  generateName: lifecycle-hook-
+spec:
+  entrypoint: main
+  hooks:
+    running:
+      expression: workflow.status == "Running"
+      template: hook
+    failed:
+      expression: workflow.status == "Failed"
+      template: hook
+
+  templates:
+    - name: main
+      steps:
+      - - name: step1
+          template: argosay
+
+    - name: argosay
+      container:
+        image: argoproj/argosay:v2
+        command: ["/argosay; exit 1"]
+
+    - name: hook
+      container:
+        image: argoproj/argosay:v2
+        command: ["/argosay"]
+`).When().
+		SubmitWorkflow().
+		WaitForWorkflow(fixtures.ToBeFailed).
+		Then().
+		ExpectWorkflow(func(t *testing.T, metadata *v1.ObjectMeta, status *v1alpha1.WorkflowStatus) {
+			assert.Equal(t, status.Phase, v1alpha1.WorkflowSucceeded)
+		}).ExpectWorkflowNode(func(status v1alpha1.NodeStatus) bool {
+			return strings.Contains(status.Name, ".hooks.running")
+		}, func(t *testing.T, status *v1alpha1.NodeStatus, pod *apiv1.Pod) {
+			assert.Equal(t, v1alpha1.NodeSucceeded, status.Phase)
+		}).ExpectWorkflowNode(func(status v1alpha1.NodeStatus) bool {
+			return strings.Contains(status.Name, ".hooks.failed")
+		}, func(t *testing.T, status *v1alpha1.NodeStatus, pod *apiv1.Pod) {
+			assert.Equal(t, v1alpha1.NodeSucceeded, status.Phase)
+		})
 }
+*/
```

Comment:
I found a new [bug](https://github.com/argoproj/argo-workflows/issues/10743) while writing above tests.
I have written test in advance in case the bug is fixed. so temporarily disable above two tests until the [issues](https://github.com/argoproj/argo-workflows/issues/10743) are resolved.

Reply:
Would you like to look into that issue too?

Reply:
@alexec i added all case of test. please check it.